android/src/main/kotlin/com/adyen/checkout/flutter/cse/AdyenCSE.kt

```diff
@@ -0,0 +1,38 @@
+package com.adyen.checkout.flutter.cse
+
+import EncryptedCardDTO
+import UnencryptedCardDTO
+import com.adyen.checkout.cse.CardEncrypter
+import com.adyen.checkout.flutter.utils.ConfigurationMapper.fromDTO
+import com.adyen.checkout.flutter.utils.ConfigurationMapper.mapToEncryptedCardDTO
+import java.lang.Exception
+
+internal class AdyenCSE {
+    fun encryptCard(
+        unencryptedCardDTO: UnencryptedCardDTO,
+        publicKey: String,
+        callback: (Result<EncryptedCardDTO>) -> Unit
+    ) {
+        try {
+            val unencryptedCard = unencryptedCardDTO.fromDTO()
+            val encryptedCard = CardEncrypter.encryptFields(unencryptedCard, publicKey)
+            val encryptedCardDTO = encryptedCard.mapToEncryptedCardDTO()
+            callback(Result.success(encryptedCardDTO))
```

Comment:
Is it intentional that you are using a callback instead of just returning a `Result<EncryptedCardDTO>`?

Reply:
This allows us to assign the encryptCard or encryptBin methods directly without additional lines in the Flutter to Kotlin communication layer. That's the [implementation](https://github.com/Adyen/adyen-flutter/blob/cc7196f7e75057ca6af24480165fe799f1214624/android/src/main/kotlin/com/adyen/checkout/flutter/CheckoutPlatformApi.kt#L75).   
However, we could introduce a return value and call the callback when the result is available. This could also simplify writing tests. I tend to change that.